### ISSUE TITLE
Question about logging events

### ISSUE THREAD

**Issue by SmutsMuts** [lloesche/valheim-server-docker#435]
Hey!

I've had some time off from the game but me and some friends started playing again and I wanted to check if you knew how to create discord events for when someone dies or logs off the server? 

I used yaml file someone created to host the container in Azure and one the options I added to get a notification when someone joins the game is the following: 
```
name: ON_VALHEIM_LOG_FILTER_CONTAINS_Spawned
value: '{ read l; l=${l//*ZDOID from /}; l=${l// :*/}; msg="Player $l spawned into matsvalheimserver1"; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$msg\"}" "$DISCORD_WEBHOOK"; }'
```

It's not perfect because this is when they spawn so will basically generate an event every time someone dies as well as logs in to the server. 
What I am after is to have an announcement when 
* Someone logs in to the server - So I know that someone is playing
* Someone quits the server - So I know if they have stopped playing

Bonus announcement would be
* Dies (because its fun and you can laugh at them :D)

I guess you can add different things as environment variables for log filters but wasn't sure what was available and thought someone maybe had some nice ideas around fun things to announce in discord based on events in the game.

**Comment by smosse** [comment]
@SmutsMuts hey I have the same env confirmation but I don't get my messages sent to my discord.
I can send a test message . like hook message work. but on log filter won't. 
Do you have a specific setting ?

**Comment by SmutsMuts** [comment]
Hey @smosse,
I will post what I have and hopefully something will help :D

I have the DISCORD_WEBHOOK variable set to a url to discord. 
Then I have VALHEIM_LOG_FILTER_CONTAINS_Spawned this one set as well to "Got character ZDOID from", think this is something I copy-pasted so not sure if it is needed or not, haven't tried without it :D 

I also have a PRE_BOOTSTRAP_HOOK but I think that one is mainly for posting the discord message when the server starts up, but might be the connection to discord as well?
Value for this one for me is: 
`curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$DISCORD_MESSAGE\"}" "$DISCORD_WEBHOOK"`

**Comment by smosse** [comment]
@SmutsMuts Yes I have the PRE_BOOTSTRAP_HOOK working but can"t get the ON_VALHEIM_LOG_FILTER_CONTAINS_Spawned working.
strangely in the log I can see it should be working..
![Capture d’écran 2022-01-12 à 11 10 46](https://user-images.githubusercontent.com/897989/149120513-16c6640b-51f1-4a38-a8be-90c5541f9826.png)
e

**Comment by smosse** [comment]
@SmutsMuts BTW for your problem. I have check the logs and when someone dies after Got character ZDOID you have a 0:0 so perhaps try with this ?
For the logout after Got character ZDOID you should see a series of 'Destroying abandoned non persistent zdo' with the player ID.

**Comment by pendrgnmmj** [comment]
Does anyone use this? I added it to my YAML for docker compose, and all the messages work to discord except the player logging in... did I do something incorrect? 

`      - DISCORD_WEBHOOK=https://discord.com/api/webhooks/825830166206611489/222w5s4ekFvHfasrCdpvZei6PIxhgZUhnxuk9F9F07Pc80tvdIqPjxtuUUC6rrBcVryp
      - DISCORD_MESSAGE=Starting Valheim server $$SERVER_NAME
      - DISCORD_MESSAGE4=Server ONLINE $$SERVER_NAME
      - DISCORD_MESSAGE2=Shutting Down server $$SERVER_NAME
      - DISCORD_MESSAGE3=Server OFFLINE $$SERVER_NAME
      - DISCORD_MESSAGE5=Server Restarting $$SERVER_NAME
      - 'PRE_BOOTSTRAP_HOOK=curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$$(eval echo $$DISCORD_MESSAGE)\"}" "$$DISCORD_WEBHOOK"'
      - 'POST_START_HOOK=curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$$(eval echo $$DISCORD_MESSAGE4)\"}" "$$DISCORD_WEBHOOK"'
      - 'PRE_SERVER_SHUTDOWN_HOOK=curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$$(eval echo $$DISCORD_MESSAGE2)\"}" "$$DISCORD_WEBHOOK"'
      - 'POST_SERVER_SHUTDOWN_HOOK=curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$$(eval echo $$DISCORD_MESSAGE3)\"}" "$$DISCORD_WEBHOOK"'
      - 'PRE_RESTART_HOOK=curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$$(eval echo $$DISCORD_MESSAGE5)\"}" "$$DISCORD_WEBHOOK"'
      - 'VALHEIM_LOG_FILTER_CONTAINS_Spawned=Got character ZDOID from '
      - 'ON_VALHEIM_LOG_FILTER_CONTAINS_Spawned=''{ read l; l=$${l//*ZDOID from /}; l=$${l// :*/}; msg="Player $$l spawned into the world"; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$$(eval echo $$msg)\"}" "$$DISCORD_WEBHOOK"; }'''
`

**Comment by pendrgnmmj** [comment]
Does anyone use this? I added it to my docker compose file and all the  messages work to discord but that player joined. Any idea what might be wrong? Thanks! 
   `- DISCORD_MESSAGE=Starting Valheim server $$SERVER_NAME`
    `- 'PRE_BOOTSTRAP_HOOK=curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$$(eval echo $$DISCORD_MESSAGE)\"}" "$$DISCORD_WEBHOOK"'
 `   
 `- 'VALHEIM_LOG_FILTER_CONTAINS_Spawned=Got character ZDOID from '`
    `- 'ON_VALHEIM_LOG_FILTER_CONTAINS_Spawned=''{ read l; l=$${l//*ZDOID from /}; l=$${l// :*/}; msg="Player $$l spawned into the world"; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$$(eval echo $$msg)\"}" "$$DISCORD_WEBHOOK";  #}'''`

**Comment by schildbach** [comment]
@pendrgnmmj I would recommend to not put these configs into the `docker-compose.yml`, but use an `.env` file instead. The reason is less escaping needed, less hassle.

Also have a look at #455.
